Bug fix/id-generator-binding

nickperry12 opened the pull request:
Fixed a bug where the ID generator was not properly binding when being passed to instances of MessageList.

Fixed an issue in the MessageList constructor where `this.generateMessageId` was only being assigned if `threadId` exists, but that should not be the case.

## Description

Small change to how `generateMessageId` was being passed to the `MessageList` class; it was losing context in it's previous implementation, returning a value of `undefined` and the ID generator was falling back to UUIDs. This change ensures it's retaining the correct context

I also revamped the `idgenerator.test.ts` test suite to cover user workflows more effectively. 

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation: #123 -->

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works

nickperry12 committed:
Bug fix

Fixed a bug where the ID generator was not properly binding when being passed to instances of MessageList.

Fixed an issue in the MessageList constructor where `this.generateMessageId` was only being assigned if `threadId` exists, but that should not be the case.

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>

nickperry12 committed:
chore: fixed lint issues

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>

DanielSLew committed:
Merge branch 'main' into bug/id-gen-binding

wardpeet commented on packages/core/src/agent/index.ts:
```diff
@@ -1452,7 +1451,7 @@ export class Agent<
           threadId,
           resourceId,
-          generateMessageId,
+          generateMessageId: this.#mastra?.generateId.bind(this.#mastra),
```
```suggestion
          generateMessageId: this.#mastra?.generateId?.bind(this.#mastra),
```

nickperry12 replied:
Gotcha, added that in. Also addressed the `streaming-memory` test failing in the latest commit. I commented out the logic for the `experimental_generateMessageId` on lines 1996-1999. I don't think that's being used, at least not at the moment. Seems like that's possibly something for a future feature where ID generators can be attached to specific primitives? The latest commit also ensures custom IDs are being used for assistant messages as opposed to IDs generated by the AI SDK

nickperry12 committed:
chore

removed unnecessary type

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>

nickperry12 committed:
Merge branch 'bug/id-gen-binding' of https://github.com/nickperry12/mastra into bug/id-gen-binding

nickperry12 committed:
Fix

Custom IDs were not being attached to assistant messages. To fix this, when retrieving the response messages from the LLM, the IDs are removed so that the custom ID generator can be called

Modified the a test in `streaming-memory` to be more consistent with the change

Co-Authored-By: Jason Jones <jasonjabarjones@gmail.com>
